assessment/report_generator_test.go

```diff
@@ -0,0 +1,2410 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package assessment
+
+import (
+	// "os"
+	// "path/filepath"
+
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/constants"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/schema"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/spanner/ddl"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetElementTypeForColumn(t *testing.T) {
+	testCases := []struct {
+		name   string
+		column utils.SrcColumnDetails
+		want   string
+	}{
+		{
+			name:   "Standard column",
+			column: utils.SrcColumnDetails{GeneratedColumn: utils.GeneratedColumnInfo{IsPresent: false}},
+			want:   "Column",
+		},
+		{
+			name:   "Generated column",
+			column: utils.SrcColumnDetails{GeneratedColumn: utils.GeneratedColumnInfo{IsPresent: true}},
+			want:   "Generated Column",
+		},
+		{
+			name:   "Zero-value column (edge case)",
+			column: utils.SrcColumnDetails{},
+			want:   "Column",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := getElementTypeForColumn(tc.column)
+			assert.Equal(t, tc.want, got)
+		})
+	}
+}
+
+func TestTableDefinitionToString(t *testing.T) {
+	testCases := []struct {
+		name               string
+		input              utils.SrcTableDetails
+		expectedSubstrings []string
+		expectedNumParts   int
+	}{
+		{
+			name: "UTF charset and multiple properties",
+			input: utils.SrcTableDetails{
+				Charset:    "utf8mb4",
+				Properties: map[string]string{"ENGINE": "InnoDB", "COMPRESSION": "ZLIB"},
+			},
+			expectedSubstrings: []string{"CHARSET=utf8mb4", "ENGINE=InnoDB", "COMPRESSION=ZLIB"},
+			expectedNumParts:   3,
+		},
+		{
+			name:               "Non-UTF charset with no properties",
+			input:              utils.SrcTableDetails{Charset: "latin1"},
+			expectedSubstrings: []string{},
+			expectedNumParts:   0,
+		},
+		{
+			name:               "Empty struct",
+			input:              utils.SrcTableDetails{},
+			expectedSubstrings: []string{},
+			expectedNumParts:   0,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tableDefinitionToString(tc.input)
+			parts := strings.Fields(got)
+			assert.Len(t, parts, tc.expectedNumParts, "The number of generated parts should match")
+
+			for _, sub := range tc.expectedSubstrings {
+				assert.Contains(t, got, sub)
+			}
+		})
+	}
+}
+
+func TestSourceColumnDefinitionToString(t *testing.T) {
+	testCases := []struct {
+		name  string
+		input utils.SrcColumnDetails
+		want  string
+	}{
+		{
+			name: "Everything is set",
+			input: utils.SrcColumnDetails{
+				Datatype:   "INT64",
+				Mods:       []int64{10, 2},
```

Comment:
nit: Since MySQL doesn't have INT64 and would have only mods , maybe change the values here to depict actual ones